lib/honeybadger/client.ex

```diff
@@ -71,11 +72,32 @@ defmodule Honeybadger.Client do
   # Callbacks
 
   def init(state) do
+    warn_if_incomplete_env(state)
```

Comment:
What currently happens if an API key is configured but the environment name isn't? I'm wondering if we still need to make `environment_name` mandatory or if this could just warn if the API key is missing.

Reply:
> What currently happens if an API key is configured but the environment name isn't?

It prints out an error `mandatory :honeybadger config key :environment_name not set`. The default value of `:environment_name` is set to `Mix.env()`, so it wouldn't be nil unless it is set to nil (https://github.com/honeybadger-io/honeybadger-elixir/blob/master/mix.exs#L30).

However, it doesn't stop sending notices. It stops sending notices only if the `api_key` is missing.

Reply:
Makes sense. Let's just roll with this.